pages/advanced/eip-7702/7702-safe.mdx

```diff
@@ -0,0 +1,59 @@
+
+Existing Safe contracts cannot be used with EIP-7702, because of following reasons:
+
+- Delegating to Safe Singleton or the Proxy contract will expose the EOA account to the risk of front-running during setup.
+- In its current implementation, the Safe Singleton contract does not let itself to become an owner meaning that after delegating to the Safe Singleton, the EOA account cannot sign Safe transactions and will need to keep another private key to do so.
+
+## Possible approaches
+
+### Modified safe proxy
+
+[SafeEIP7702Proxy](https://github.com/5afe/safe-eip7702/blob/main/safe-eip7702-contracts/contracts/SafeEIP7702Proxy.sol)
+
+This approach uses a derived proxy contract from [Safe Proxy](https://github.com/5afe/safe-eip7702/blob/main/safe-eip7702-contracts/contracts/SafeEIP7702Proxy.sol) with following changes:
+1. The constructor of the `SafeEIP7702Proxy` contract has an additional parameter `setupDataHash` which is the hash of the `setup` function call data. Thus, the address of the proxy contract also depends on the `setupDataHash` and not just the `singleton` contract address.
+Proxy contract uses this hash to verify that the `setup` function parameter values are unchanged during initialised of storage.
+
+2. The proxy implements `setup` function which calls the `setup` function of the `singleton` contract and has additional logic:
+- Set the storage slot 0 that is, the address of the singleton in the EOA storage.
+- Verify that the `setupDataHash` is equal to the hash of the `setup` function call data.
+
+This approach has a gas overhead as a new proxy contract has to be deployed for each EOA account, as the `setupDataHash` may be unique for each user. However, using this approach, users can use Safe\{Wallet\} with minor modifications and import the EOA account as a Safe account. 
+
+![diagram-7702-approach-1](../../../assets/diagram-7702-approach-1.png)
+
+Follow the instructions here to use this approach to set code in EOA: https://github.com/5afe/safe-eip7702/tree/main/safe-eip7702-contracts#execute-scripts
+
+### Modified safe singleton
+
+In this approach, a derived version of Safe Singleton, aka `SafeEIP7702`, overrides the `setup` function and reverts when called. Instead, the contract's new `setupEIP7702` function has a `signature` parameter. The default owner will be set to the address of the EOA account that delegates to this Singleton contract with a threshold of 1.
+
+[SafeEIP7702](https://github.com/safe-global/safe-smart-account/blob/feature/eip-7702/contracts/experimental/SafeEIP7702.sol)
+
+Because of no proxy contract in this approach, the storage slot 0 remains unused. Safe Transaction Service and other services that depend on the value at storage slot 0 will not work with this approach.
```

Comment:
```suggestion
Because this approach does not use a proxy contract, storage slot 0 remains unused. The Safe Transaction Service and other services that depend on the value at storage slot 0 will not work with this approach.
```